Make smoothing optional and fix some pep8 issues

havatv opened the pull request:
Make smoothing optional and fix some pep8 issues.

Fix #15

mkondratyev85 commented on pointtool.py:
```diff
@@ -59,50 +60,49 @@ def snap_tolerance_changed(self, snap_tolerance):
-                    level=Qgis.Critical, duration=2)
+            self.iface.messageBar().pushMessage("Missing Layer",
+                            "Can't trace indexed or gray image",
+                   d              level=Qgis.Critical, duration=2)
```
Please remove 'd' character from the beginning of the line.

mkondratyev85 commented:
I agree with the proposed changes. 

Before I merge this pull request I want to clarify a few things.

1. Please fix typo in pointtool.py on line 113

2. What is the purpose of the combobox right to the checkbox you added?
Right now it will confuse users, i'm afraid.

![Снимок экрана от 2020-09-03 10-42-04](https://user-images.githubusercontent.com/32978900/92048609-d1163f80-edd3-11ea-8cfa-7a4d008ae9be.png)

mkondratyev85 commented on raster_tracer_dockwidget_base.ui:
```diff
@@ -69,6 +69,16 @@
+      </property>
+     </widget>
+    </item>
+    <item row="6" column="1">
+     <widget class="QComboBox" name="comboBox"/>
+    </item>
```
What is the purpose of this comboBox?

havatv commented:
> Before I merge this pull request I want to clarify a few things.
>     1. Please fix typo in pointtool.py on line 113

Sorry about that! Fixed now.

>     2. What is the purpose of the combobox right to the checkbox you added?
>        Right now it will confuse users, i'm afraid.

The plan was to let the user choose smoothing method (smoothing / simplification, Douglas-Peucker, Chaiken, Snake, ...). But that can wait. I have removed it.

mkondratyev85 commented:
> The plan was to let the user choose smoothing method (smoothing / simplification, Douglas-Peucker, Chaiken, snakes, ...). But that can wait. I have removed it.


Ok. I wasn't sure if it was intentional or not. Added an issue for that - #17

havatv commented:
Thanks for handling this so soon, @mkondratyev85.
We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)

mkondratyev85 commented:
> We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)

Sure. What other futures do you want to see in `raster_tracer` before the release? And what this course will be about?

havatv commented:
> > We are planning to use *Raster Tracer" in course this autumn, so it would be nice to have a new release og the plugin (with this feature included) in the not too distant future... :-)
> 
> Sure. What other futures do you want to see in `raster_tracer` before the release? And what this course will be about?

Thanks a lot for being so helpful, @mkondratyev85.
I think everything should be fine now. I wanted the plugin to be easy to use for students. The geometry type warning should be helpful. In addition, optional smoothing will help them understand how the tracing is working.

The course is a basic GIS course here at NMBU. Thanks to the *Raster Tracer* plugin, it is now possible for the students to only use QGIS for the exercises.This is particularly useful in these Covid19 times when access to the University computer laboratories is limited. Until last year, they had to use ArcScan for one of the exercises.

Thanks again!
